layouts/labs/list.html

```diff
@@ -43,81 +43,84 @@ <h3 class="title is-5">In diesen Städten gibt es bereits ein OK Lab:</h3>
-  .then(function (d) {
-
-  // draw points
-  d3.csv("{{ "js/lab-locations.csv" | relURL }}")
-  .then(function(d) {
-    console.log(d);
-  svg.selectAll("circle")
-     .data(d)
-     .enter()
-     .append("circle")
-     .attr("r", 5)
-     .style("fill", "#01a8a8")
-     .classed("pin", true)
-     .attr("cx", function (d) {
-         return projection([+d.long, +d.lat])[0];
-      })
-     .attr("cy", function (d) {
-         return projection([+d.long, +d.lat])[1];
-      })
-      .on("mouseover", d => {
+    function on_mouseout(){
+      var t = d3.transition().duration(200);
       div
-        .transition()
-        .duration(200)
-        .style("opacity", 0.9);
+        .transition(t)
+        .style("opacity", 0);
+    }
+  
+    function on_mouseover(data){
+      var t = d3.transition().duration(500);
       div
-        .html(d.city)
-        .style("left", d3.event.pageX + "px")
-        .style("top", d3.event.pageY - 28 + "px");
-    })
-    .on("click", function(d) {window.location = "https://codefor.de/" +d.city})
-    .on("mouseout", () => {
+        .html(data.city)
+        .style("left", d3.event.pageX + 8 + "px")
+        .style("top", d3.event.pageY - 43 + "px")
+        .style("pointer-events", "none");
       div
-        .transition()
-        .duration(500)
-        .style("opacity", 0);
-    });
-   });
+        .transition(t)
+        .style("opacity", 0.9);
 
-    // draw map
-    svg.append("g")
-      .selectAll("path")
-      .data(d.features)
-      .enter()
-      .append("path")
-      .attr("d", d3.geoPath().projection(projection))
-      .attr("fill", "lightgrey")
-      .style("opacity", 0.5)
-      .style("stroke","darkgrey");
-      });
+    }
+  
+    function locations_loaded(locations) {
+      console.log(locations);
+      svg.selectAll("circle")
+        .data(locations)
+        .enter()
+        .append("circle")
+        .attr("r", 5)
+        .style("fill", "#01a8a8")
+        .classed("pin", true)
+        .attr("cx", locations => { return projection([+locations.long, +locations.lat])[0]; })
+        .attr("cy", locations => { return projection([+locations.long, +locations.lat])[1]; })
+        .on("mouseover", on_mouseover)
+        .on("click", locations => { window.location = "https://codefor.de/" + locations.city })
+        .on("mouseout", on_mouseout);
+    }
+  
+    function data_loaded(data) {
+      // draw points
+      d3.csv("{{ "js/lab-locations.csv" | relURL }}")
+      .then(locations_loaded);
+  
+      // draw map
+      svg.append("g")
+        .selectAll("path")
+        .data(data.features)
+        .enter()
+        .append("path")
+        .attr("d", d3.geoPath().projection(projection))
+        .attr("fill", "lightgrey")
+        .style("opacity", 0.5)
+        .style("stroke", "darkgrey");
+    }
 
-  } // end buildMap
+    // load data
+    d3.json("https://raw.githubusercontent.com/isellsoap/deutschlandGeoJSON/master/2_bundeslaender/4_niedrig.geojson")
```

Comment:
Kannst du die Datei mit hier ins Repository ziehen? Und am besten an irgend einer Stelle auch die Quelle angeben?

Reply:
Die Adresse habe ich nur aus dem original Code übernommen.

Wäre es sinnvoll das entsprechende Repo als Git-Submodule zu laden? Ansonsten wäre es reine Duplikation ohne Möglichkeiten zum Update. Vll gibt es aber auch stabile Release-Branches, wäre das in Ordnung?

Die Referenz zum Ursprung kann man sicherlich noch einfügen.

Reply:
Nein, kein submodule. Einfach dass die Datei diesem Repository geladen wird. Ruhig duplizieren, aber bitte wie gesagt die Quelle irgendwo einfügen

Reply:
Die Datei ist ins Repo in den Ordner static/labs gelegt worden.
Ausserdem habe ich eine Quellenangabe unter der Karte eingefuegt.